airbyte-ci/connectors/live-tests/src/live_tests/conftest.py

```diff
@@ -906,3 +927,23 @@ async def read_with_state_target_execution_result(
     request.config.stash[stash_keys.REPORT].add_target_execution_result(execution_result)
 
     return execution_result
+
+
+def respect_validation_test_mode(func: Callable) -> Any:
```

Comment:
❓ Would it be worth while to break out the "Test mode" logic into its own file? e.g. `config/test_modes.py` ?

Reply:
Sure, that feels cleaner. Done.